File: module_network/src/main/java/com/zch/network/retrofithelper/interceptor/HeaderInterceptor.java
package com.zch.network.retrofithelper.interceptor;


import com.zch.network.retrofithelper.RetrofitHelper;

import java.io.IOException;
import java.util.Map;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class HeaderInterceptor implements Interceptor {
    @Override
    public Response intercept(Chain chain) throws IOException {
        //如果支持添加头
        if(RetrofitHelper.getInstance().isAddHeader()){

            Map<String,String> headers = RetrofitHelper.getInstance().getHeaders();
            if(!headers.isEmpty()){
                Request.Builder builder = chain.request().newBuilder();
                for(Map.Entry<String,String> entry : headers.entrySet()){
                    builder.addHeader(entry.getKey(),entry.getValue());
                }
                return chain.proceed(builder.build());
            }
        }

        return chain.proceed(chain.request());
    }
}

File: app/src/main/java/com/zch/basedemo/network/network/HeaderInterceptor.kt
package com.zch.basedemo.network.network

import okhttp3.Interceptor
import okhttp3.Response

/**
 *
 * Description: 请求头
 * Created by zhangchenhan on 2022/01/11 5:41 下午
 *
 **/
class HeaderInterceptor : Interceptor {
    override fun intercept(chain: Interceptor.Chain): Response {
        val originalRequest = chain.request()
        val authorised = originalRequest.newBuilder().build()
        return chain.proceed(authorised)

    }
}